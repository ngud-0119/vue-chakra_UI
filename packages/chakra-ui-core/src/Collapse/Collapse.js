import { CAnimateHeight } from '../Transition'
import CBox from '../Box'
import { forwardProps } from '../utils'

const CCollapse = {
  name: 'CCollapse',
  props: {
    isOpen: Boolean,
    duration: {
      type: Number,
      default: 250
    },
    easing: {
      type: String,
      default: 'easeInOutSine'
    },
    startingHeight: Number,
    endingHeight: Number,
    animateOpacity: {
      type: Boolean,
      default: true
    }
  },
  render (h) {
    const children = this.$slots.default

    return h(CAnimateHeight, {
      props: {
        isOpen: this.isOpen,
        duration: this.duration,
        enterEasing: this.easing,
        leaveEasing: this.easing,
        initialHeight: this.startingHeight,
        finalHeight: this.endingHeight,
        animateOpacity: this.animateOpacity
      },
      on: {
        enter: (e) => this.$emit('start', e),
        leave: (e) => this.$emit('finish', e)
      }
    }, [h(CBox, {
      props: {
        ...forwardProps(this.$props),
        overflow: 'hidden'
      }
    }, children)])
  }
}

export default CCollapse
