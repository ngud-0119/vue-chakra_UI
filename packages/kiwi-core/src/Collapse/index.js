import { AnimateHeight } from '../Transition'
import Box from '../Box'

const Collapse = {
  name: 'Collapse',
  props: {
    isOpen: Boolean,
    duration: {
      type: Number,
      default: 250
    },
    easing: {
      type: String,
      default: 'easeInOutSine'
    },
    startingHeight: Number,
    endingHeight: Number,
    animateOpacity: {
      type: Boolean,
      default: true
    }
  },
  render (h) {
    const children = this.$slots.default

    return h(AnimateHeight, {
      props: {
        isOpen: this.isOpen,
        duration: this.duration,
        enterEasing: this.easing,
        leaveEasing: this.easing,
        initialHeight: this.startingHeight,
        finalHeight: this.endingHeight,
        animateOpacity: this.animateOpacity
      },
      on: {
        enter: (e) => this.$emit('start', e),
        leave: (e) => this.$emit('finish', e)
      }
    }, [h(Box, { props: { overflow: 'hidden' } }, children)])
  }
}

export default Collapse
