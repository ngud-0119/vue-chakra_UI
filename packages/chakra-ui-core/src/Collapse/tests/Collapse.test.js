import Collapse from '../'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { Collapse },
    template: `<Collapse ${inlineAttrs}>content</Collapse>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})
