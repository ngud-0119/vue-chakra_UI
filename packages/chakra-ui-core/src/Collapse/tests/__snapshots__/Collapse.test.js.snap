// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-i6bazn"
  >
    <div
      class="css-i6bazn"
    >
      content
    </div>
  </div>
</DocumentFragment>
`;
